src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/PermissionsManager.java

```diff
@@ -170,16 +172,20 @@ public List<String> getPermissions(Clan clan) {
      * Charge a player some money
      *
      */
-    public boolean playerChargeMoney(OfflinePlayer player, double money) {
-        return economy.withdrawPlayer(player, money).transactionSuccess();
+    public boolean playerChargeMoney(OfflinePlayer player, double money, Cause cause) {
+        SimpleClansEconomyTransactionEvent event = new SimpleClansEconomyTransactionEvent(player, money, cause, SimpleClansEconomyTransactionEvent.TransactionType.WITHDRAW_FROM_PLAYER);
+        Bukkit.getPluginManager().callEvent(event);
+        return economy.withdrawPlayer(player, event.getAmount()).transactionSuccess();
     }
 
     /**
      * Grants a player some money
      *
      */
-    public boolean playerGrantMoney(OfflinePlayer player, double money) {
-        return economy.depositPlayer(player, money).transactionSuccess();
+    public boolean playerGrantMoney(OfflinePlayer player, double money, Cause cause) {
```

Comment:
Hi, firstly, thanks for your PR. 😃 
We like when new people help us to maintain SimpleClans.
However, there is a _small_ issue.
 
You made a breaking change to API (changed the signature of `PermissionManager#playerGrantMoney` and `#playerChargeMoney`).
Imagine if other developers right now using this method (I want to believe they aren't, but we can't know), these changes would break their plugins (it's not good, right?).

I think, you may mark the older method as `@Deprecated(since = "2.19.0", forRemoval = true)`, then `Cause` in the SimpleClansEconomyTransactionEvent's constructor may be marked as `@Nullable`. I've seen you used `CUSTOM` for deprecated methods ([one](https://github.com/RoinujNosde/SimpleClans/pull/384/files#diff-ed5554542c2aa50887a4ace2e62e8aee7793e2109ffd716de28f0ccbae1bd4d9R116) and [second](https://github.com/RoinujNosde/SimpleClans/pull/384/files#diff-ed5554542c2aa50887a4ace2e62e8aee7793e2109ffd716de28f0ccbae1bd4d9R153)) (mark them `forRemoval` by the way or even delete them if they aren't used anywhere), but I prefer don't see any cause if it's called from the deprecated method.

Also remove `SimpleClans` from `SimpleClansEconomyTransactionEvent`. 
And rename your PR to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).

Reply:
You're totally right!

I've changed my PR with the convention, removed both deprecated methods since there was no callings. I've added the old method and marked it as Deprecated with the tag. and I think the issue is now fixed!